### ISSUE TITLE
What will RDChiral do when the template is adjusted to many reaction certers?

### ISSUE THREAD

**Issue by Jourllker** [connorcoley/rdchiral#49]
What a awesome work!!!
But I found that when using rdchiral.main.rdchiralRunText it only select a random reaction center if there are not only one reaction center adjusting the template. And the code and git seem do not refer to this, so what can i do if i want the result of all reaction centers when my template adjusting many reaction centers? Thanks!!!

**Comment by connorcoley** [comment]
Can you provide a specific example where this is happening?

**Comment by Jourllker** [comment]
For a single example, the template is '[C;H0;D3;+0:1]-[N;H1;D2;+0:3]>>[C;H0;D3;+0:1]-[Cl;H0;D1;+0].[N;H2;D1;+0:3]' and the reactant is "O=C(C1=CC=CC(C(F)(F)F)=C1)NC2=CC=CC(C(N(CC3)CN3C4=C(C=CN5)C5=NC=C4)=O)=C2". The template wants to break the C-N bond, and there are two C-N bonds in the reactant.
![QQ图片20241112154528](https://github.com/user-attachments/assets/4c67abc5-5982-4fcd-859c-d32eda7c0efe)

**Comment by connorcoley** [comment]
It looks like there's only one site that should match your template since it requires a secondary nitrogen, not a tertiary nitrogen (i.e., `[N;H1;D2;+0:3]`)

**Comment by Jourllker** [comment]
Yeah, you are right. My example can not explain the issue. 
If the molecule just like this, what will we get if use the template? 
![QQ图片20241112184616](https://github.com/user-attachments/assets/40fc3c41-6599-4b00-a9ac-ae10b7813ff5)

**Comment by Jourllker** [comment]
Yeah, you are right. My example can not explain the issue.
If the molecule just like this, what will we get if use the template? The reaction center is in 1 or 2?
![Uploading QQ图片20241112184616.png…]()

**Comment by Jourllker** [comment]
Yeah, you are right. My example can not explain the issue.
If the molecule just like this, what will we get if use the template? The reaction center is in 1 or 2?
![QQ图片20241112184616](https://github.com/user-attachments/assets/d37e3420-0f1b-479a-977c-daec0ebf76a6)

**Comment by connorcoley** [comment]
The routine for applying a template should return a list of outcomes in that case. In that specific case, however, I believe symmetry would mean thaht those two outcomes are collapsed into a single one. If you break the symmetry of the molecule and apply the template, you should see multiple outcomes

**Comment by Jourllker** [comment]
Yes, you are right. I have tried this that the rdchiralRunText returns a list. When there is only one site, the list contains only one element. If there are other sites, the list contains not only one element.
I'm so grateful for your help!
